### PR TITLE
Wanl/serverless sdk

### PR DESCRIPTION
**Pull request by realwanpengli**
Serverless SDK, supports all rest api v1.0

### CODE REVIEW

File: NuGet.config [hunk 1]
```diff
@@ -1,7 +1,5 @@
 ﻿<?xml version="1.0" encoding="utf-8"?>
 <configuration>
   <packageSources>
-    <clear />
-    <!-- Restore sources should be defined in build/sources.props. -->
-  </packageSources>
+    <clear /></packageSources>
```
Comment (vicancy) [inline NuGet.config]: undo the change

File: AzureSignalR.sln [hunk 2]
```diff
@@ -48,7 +48,11 @@ Project("{9A19103F-16F7-4668-BE54-9A1E7A4F7556}") = "AspNet.ChatSample.SelfHostS
 EndProject
 Project("{9A19103F-16F7-4668-BE54-9A1E7A4F7556}") = "AspNet.ChatSample.CSharpClient", "samples\AspNet.ChatSample\AspNet.ChatSample.CSharpClient\AspNet.ChatSample.CSharpClient.csproj", "{38522915-6EA5-4F1D-B6EC-7FE4B178F579}"
 EndProject
-Project("{FAE04EC0-301F-11D3-BF4B-00C04F79EFBC}") = "AspNet.ChatSample.JavaScriptClient", "samples\AspNet.ChatSample\AspNet.ChatSample.JavaScriptClient\AspNet.ChatSample.JavaScriptClient.csproj", "{5C5D6C13-61BA-48B8-803A-E6A03D31499E}"
+Project("{9A19103F-16F7-4668-BE54-9A1E7A4F7556}") = "AspNet.ChatSample.JavaScriptClient", "samples\AspNet.ChatSample\AspNet.ChatSample.JavaScriptClient\AspNet.ChatSample.JavaScriptClient.csproj", "{5C5D6C13-61BA-48B8-803A-E6A03D31499E}"
+EndProject
+Project("{9A19103F-16F7-4668-BE54-9A1E7A4F7556}") = "Microsoft.Azure.SignalR.ServerlessAgent", "src\Microsoft.Azure.SignalR.ServerlessAgent\Microsoft.Azure.SignalR.ServerlessAgent.csproj", "{2D1E4D0E-0F24-41D5-88C0-82F872E95DF6}"
```
Comment (vicancy) [inline AzureSignalR.sln]: Consider a better naming

File: src/Microsoft.Azure.SignalR.ServerlessAgent/AccessToken.cs [hunk 3]
```diff
@@ -0,0 +1,36 @@
+﻿using System;
```
Comment (vicancy) [inline src/Microsoft.Azure.SignalR.ServerlessAgent/AccessToken.cs]: copy right header

File: src/Microsoft.Azure.SignalR.ServerlessAgent/AccessToken.cs [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+using System.Collections.Generic;
+using System.Text;
+using System.IdentityModel.Tokens.Jwt;
+using System.Security.Claims;
+using Microsoft.IdentityModel.Tokens;
+
+namespace Microsoft.Azure.SignalR.ServerlessAgent
+{
+    public class AccessToken
```
Comment (vicancy) [inline src/Microsoft.Azure.SignalR.ServerlessAgent/AccessToken.cs]: For SDKs, always use `internal` unless you'd like the class to be used by SDK consumers

File: AzureSignalR.sln [hunk 5]
```diff
@@ -48,7 +48,11 @@ Project("{9A19103F-16F7-4668-BE54-9A1E7A4F7556}") = "AspNet.ChatSample.SelfHostS
 EndProject
 Project("{9A19103F-16F7-4668-BE54-9A1E7A4F7556}") = "AspNet.ChatSample.CSharpClient", "samples\AspNet.ChatSample\AspNet.ChatSample.CSharpClient\AspNet.ChatSample.CSharpClient.csproj", "{38522915-6EA5-4F1D-B6EC-7FE4B178F579}"
```
Comment (vicancy) [inline AzureSignalR.sln]: This is a big PR, consider separate it into several small ones and each has a clear description. One suggestion is, one PR has one simple and clear target, one problem to solve

File: src/Microsoft.Azure.SignalR.ServerlessAgent/RestHubLifetimeManager.cs [hunk 6]
```diff
@@ -0,0 +1,134 @@
...
+    {
+        public RestHubLifetimeManager(AgentContext agentContext)
+        {
+            _context = agentContext;
+            _apis = new RestApis(_context.GetEndpoint(), _context.HubName, _context.RestApiVersion);
+        }
+
+        public override Task AddToGroupAsync(string connectionId, string groupName, CancellationToken cancellationToken = default)
+        {
+            throw new NotImplementedException();
```
Comment (vwxyzh) [inline src/Microsoft.Azure.SignalR.ServerlessAgent/RestHubLifetimeManager.cs]: `NotSupportedException`

**Comment by realwanpengli** [comment]
This PR will be separated into small pieces to be clear.
